typescript/lib/services/container_creation_config.ts

```diff
@@ -0,0 +1,122 @@
+import { Result } from "neverthrow";
+
+const DEFAULT_KURTOSIS_VOLUME_MOUNTPOINT: string = "/kurtosis-enclave-data";
+
+export type StaticFileID = string;
+
+// The ID of an artifact containing files that should be mounted into a service container
+export type FilesArtifactID = string;
+
+// ====================================================================================================
+//                                    Config Object
+// ====================================================================================================
+// TODO defensive copy when we're giving back complex objects?????
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+export class ContainerCreationConfig {
+	
+    private readonly image: string;
+    private readonly kurtosisVolumeMountpoint: string; // Technically the enclave data volume, but we call it this for simplicity for the user
+    private readonly usedPortsSet: Map<string, boolean>;
+    private readonly fileGeneratingFuncs: Map<string, (fp: number) => Result<null, Error>>; // File descriptors are just integers
+    private readonly usedStaticFilesSet: Map<StaticFileID, boolean>;
+    private readonly filesArtifactMountpoints: Map<FilesArtifactID, string>;
+
+    constructor(
+        image: string,
+        kurtosisVolumeMountpoint: string,
+        usedPortsSet: Map<string, boolean>,
+        fileGeneratingFuncs: Map<string, (fp: number) => Result<null, Error>>,
+        usedStaticFilesSet: Map<StaticFileID, boolean>,
+        filesArtifactMountpoints: Map<FilesArtifactID, string>
+    ){
```

Comment:
IMO this method is also fine, if you want to leave args at 1 indendentation level rather than doing 2: it makes clear the distinction between args & code